flexget/components/sites/sites/passthepopcorn.py

```diff
@@ -262,27 +137,30 @@ def search(self, task, entry, config):
 
         search_strings = entry.get('search_strings', [entry['title']])
 
+        request_headers = {"ApiUser": config['apiuser'], "ApiKey": config['apikey']}
+
         # searching with imdb id is much more precise
         if entry.get('imdb_id'):
             search_strings = [entry['imdb_id']]
+        else:
+            # use the movie year if available to improve search results.
+            if 'movie_year' in entry:
+                params['year'] = int(entry['movie_year'])
 
         for search_string in search_strings:
             params['searchstr'] = search_string
             logger.debug('Using search params: {}', params)
             try:
-                result = self.get(
-                    self.base_url + 'torrents.php',
-                    params,
-                    config['username'],
-                    config['password'],
-                    config['passkey'],
-                ).json()
+                siteresponse = requests.get(
```

Comment:
Should we be using `task.requests` here, to make sure plugins modifying the session work?

Reply:
You tell me? Mostly just replicating the existing style, calls of the original plugin and similar site plugins. If you think this is of high value then it... "looks" simple enough to test.

One thing I do have a consern about / may need more testing is that I would like the plugin to try and follow enforce rate limiting rules and not let the user over ride them. At the moment that is just the same domain request per second the old plugin (but with fewer calls due to no cookie login loop) had but it may need a further PR with more restrictions if anything is discovered after heavier use.

Reply:
I think we should be using it. Other plugins may not be, because I think originally the search api didn't have access to the task. The good thing about using `task.requests` is that the user could customize the limiting behavior by using the domain_delay plugin.

Reply:
Do they publish what the actual limit is?

Reply:
> Do they publish what the actual limit is?

No, from discussions on the forum for other scripts it appears they don't want to disclose the exact values / methods they are using to detect abuse. In general it was stated that no faster than 1 request per second and HALT your script if you get an HTTP error. (no specifics on any errors are given).

Since removing the old login logic I have not actually hit the rate limit with the new code as it really only makes one request per title and waits 5s. The old code would hammer the auth URL if the cookies didn't work, and with the new abuse filters the old code will auto lock you out of the site quite quickly.

Reply:
change commited.

Reply:
This seems to have removed the rate limit completely, I was just doing a more extensive stress test on the plugin and noticed it was not rate limiting at all now.  I am guessing as the rate limit is not applied to the correct object now.. looking into it.

Reply:
Verified a fix to correctly attach the limit to the task.requests, it is working now.

Reply:
Hmm. Something I wondered while looking at this change. If there are 2 limiters on the same site, should they both apply? Right now only the first one does. I think at a minimum it should be changed so that the most recent applies rather than the oldest. (i.e. If the plugin includes a limiter, as in this case, one the user adds via domain_delay should override.) Potentially we should go further and just have _both_ of them apply. If they were both just timers between requests it would mean the longer wait always applied, but if a regular timed limiter was combined with a token bucket limiter that might actually be useful.

Reply:
I don't think so, pretty sure it's basically just a dict that gets overwritten since the hostname is the index. And the TimedLimiter essentially calls the TokenLimiter with a token count of 1, so they're actually the same

Reply:
Testing the change as is the coded delay overrides the users domain_delay: if they have provided one.

If you want to provide your own a fully working code snippet example for me to use instead that is fine, however this is a problem with ALL existing plugins that call add_domain_limiter inside them. 

There were two main goals in my rewrite.. replace the auth and ensure a safe default rate limit was in place..

I think those are met.. 

There are going to be a select few that use this plugin anyway.

Reply:
@bagobones Yes, sorry, I didn't mean you should do something here. I was just talking the opportunity to discuss the desired behavior

Reply:
> @bagobones Yes, sorry, I didn't mean you should do something here. I was just talking the opportunity to discuss the desired behavior

Thanks @gazpachoking , I am focused on trying to get this fix merged and I am happy to correct or implement improvements, but keep in mind this is 100% hobby time for me and I am not a full time coder.. I abandon my ombi pull request a few years ago as it seemed to be stuck in code review (might try and help with the new ombi_list one that is ALSO stuck in review).

Hoping this fix can get merged if it isn't going to create issues and helps others that want to use it.

Reply:
For posterity, I ended up changing this after merge to explicitly specify not overwriting existing limiters. I changed domain_delay to explicitly overwrite existing ones, so it should be that this (and other plugins adding their own limiter) can be properly overridden by the user domain_delay now.